ui-components/src/helpers/filters.ts

```diff
@@ -1,70 +1,217 @@
+        preferredUnitOptions
+      ),
+      new FrontEndFilter(ListingFilterKeys.zipcode, EnumListingFilterParamsComparison["IN"]),
+      new FrontEndFilter(ListingFilterKeys.minRent, EnumListingFilterParamsComparison[">="]),
+      new FrontEndFilter(ListingFilterKeys.maxRent, EnumListingFilterParamsComparison["<="]),
+      new FrontEndFilter(ListingFilterKeys.seniorHousing, EnumListingFilterParamsComparison["NA"]),
+      // Check if the filters below are used or should be deleted
+      new FrontEndFilter(ListingFilterKeys.name, EnumListingFilterParamsComparison["="]),
+      new FrontEndFilter(ListingFilterKeys.leasingAgents, EnumListingFilterParamsComparison["="]),
+      new FrontEndFilter(ListingFilterKeys.status, EnumListingFilterParamsComparison["="]),
+      new FrontEndFilter(ListingFilterKeys.ami, EnumListingFilterParamsComparison["NA"]),
+    ].reduce(function (filters: Record<string, FrontEndFilter>, currFilter: FrontEndFilter) {
+      filters[currFilter.name] = currFilter
+      return filters
+    }, {})
+  }
+}
+
+export class CommunityTypeFilter extends FrontEndFilter {
+  getBackendFilterType() {
+    return this.value
+  }
+  getBackendFilterValue() {
+    if (this.value == EMPTY_OPTION || this.value == undefined) {
+      return undefined
+    } else {
+      return true
     }
   }
 }
 
-export function encodeToBackendFilterArray(filterParams: ListingFilterParams) {
+export const blankFrontEndFilters = () => {
+  return new FrontEndFilters()
+}
+
+export function encodeToBackendFilterArray(filters: Record<string, FrontEndFilter>) {
   const filterArray = []
-  for (const filterType in filterParams) {
-    if (filterType in ListingFilterKeys) {
-      const comparison = getComparisonForFilter(ListingFilterKeys[filterType])
-      filterArray.push({
-        $comparison: comparison,
-        [filterType]: filterParams[filterType],
-      })
+  for (const filterName in filters) {
+    const type = filters[filterName].getBackendFilterType()
+    const value = filters[filterName].getBackendFilterValue()
+    const comparison = filters[filterName].comparison
+
+    if (type in ListingFilterKeys && value !== undefined && value !== "") {
+      filterArray.push({ $comparison: comparison, [type]: value })
     }
   }
   return filterArray
 }
 
-export function encodeToFrontendFilterString(filterParams: ListingFilterParams) {
+export function encodeToFrontendFilterString(filters: Record<string, FrontEndFilter>) {
   let queryString = ""
-  for (const filterType in filterParams) {
-    const value = filterParams[filterType]
-    if (filterType in ListingFilterKeys && value !== undefined && value !== "") {
-      queryString += `&${filterType}=${value}`
+  for (const filterName in filters) {
+    const type = filters[filterName].getBackendFilterType()
+    const value = filters[filterName].getBackendFilterValue()
+    if (type in ListingFilterKeys && value !== undefined && value !== "") {
```

Comment:
Instead of this, I added a property `hasBackendFilter` and used that to make sure that the communityType filter is not used in the creation of the URL or backend query.

I also updated the getBackendFilterType and getBackendFilterValue for the front end communityType filter to throw an error to make sure it's not used.